aten/src/ATen/native/Unique.cpp

```diff
@@ -135,24 +135,23 @@ ForwardIt _unique_dim_cpu_impl(ForwardIt first, ForwardIt last,
     if (first == last) {
       return last;
     }
-    // save to calculate distance to iterators
-    ForwardIt begin = first;
 
-    // set first inverse index and count
-    inverse_indices_vec[indices[0]] = 0;
-    counts[0] += 1;
+    int64_t *indices_data = indices.data();
+    int64_t *inverse_data = inverse_indices_vec.data_ptr<int64_t>();
```

Comment:
Can you please add a TORCH_INTERNAL_ASSERT that inverse_indices_vec and counts are contiguous? They currently are at the only callsite, but this function should explicitly check its assumptions.

Reply:
Sure, added.